### PR TITLE
add endpoint to get openstack networks

### LINKED ISSUE #1449: Offer endpoint for getting Network from Openstack
### Current state
`Network` is an optional field in the UI, but the user has to add it manually.

### Desired state
As the kubermatic api is already able to get all Networks, there should be an endpoint for it, so that a user could select it from a dropdown in the UI.
https://github.com/kubermatic/kubermatic/blob/master/api/pkg/provider/cloud/openstack/helper.go

### PR DESCRIPTION
**Pull request by kgroschoff**
**What this PR does / why we need it**:
add endpoint to get openstack networks

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1449 

**Special notes for your reviewer**:
/

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: api/pkg/handler/routes_v1.go [hunk 1]
```diff
@@ -361,14 +365,36 @@ func (r Routing) listOpenstackSizes() http.Handler {
...
+// swagger:route GET /api/v1/openstack/networks openstack listOpenstackNetworks
+//
+// Lists networks from openstack
+//
+//     Produces:
+//     - application/json
+//
+//     Responses:
+//       default: errorResponse
+//       200: []OpenstackNetwork
```
Comment (p0lyn0mial) [inline api/pkg/handler/routes_v1.go:390]: this doesn't fail ? 
nice !
I had to create dedicated `...List` types.

File: api/pkg/provider/cloud/openstack/provider.go [hunk 2]
```diff
@@ -262,6 +262,36 @@ func (os *Provider) GetTenants(cloud *kubermaticv1.CloudSpec) ([]apiv1.Openstack
 	return apiTenants, nil
 }
 
+// GetNetworks lists all available networks for the given CloudSpec.DatacenterName
+func (os *Provider) GetNetworks(cloud *kubermaticv1.CloudSpec) ([]apiv1.OpenstackNetwork, error) {
```
Comment (p0lyn0mial) [inline api/pkg/provider/cloud/openstack/provider.go]: I would prefer not to return `api` types from `openstack` provider or any other. I'd prefer to do the translation in the handler. But looking at the `openstack` interface seems like it is already leaking `api` types :( We are more consistent  with the existing interface when returning `api` types.

**Review by p0lyn0mial — DISMISSED** [review]

File: api/pkg/handler/routes_v1.go (hunk 1 above)
Comment (kgroschoff) [reply api/pkg/handler/routes_v1.go:390]: Yes, used to do it with `...List` as well, but `verify-swagger.sh` warned that `OpenstackNetworkList` isn't used, so I tried it like this and it works. Also saw it for `MasterVersion` 😄

File: api/pkg/provider/cloud/openstack/provider.go (hunk 2 above)
Comment (kgroschoff) [reply api/pkg/provider/cloud/openstack/provider.go]: I could try to rewrite it, if you want?

Comment (mrIncompetent) [reply api/pkg/provider/cloud/openstack/provider.go]: Yes please. The cloud provider should not be aware of the API types.
The consumer of this function is responsible for converting it to the representation it needs.

**Review by kgroschoff — COMMENTED** [review]
@mrIncompetent @p0lyn0mial is this ok? https://github.com/kubermatic/kubermatic/pull/1505/commits/5c14d38b62bdee230099e47d2320ad50ec6546a9

**Review by mrIncompetent — DISMISSED** [review]

**Review by mrIncompetent — APPROVED** [review]

**Comment by p0lyn0mial** [comment]
lgtm

@kgroschoff  thanks for updating the `openstack` interface.

**Review by p0lyn0mial — APPROVED** [review]

**Pull request merged**